deepcell/data/tracking_test.py

```diff
@@ -0,0 +1,179 @@
+
+
+@parameterized.named_parameters([
+    {
+        'testcase_name': 'test_data_01',
+        'time': 30,
+        'max_cells': 25,
+        'crop_size': 32,
+        'batch_size': 32,
+        'seed': None,
+        'track_length': 8,
+        'val_split': 0.4
+    }, {
+        'testcase_name': 'test_data_02',
+        'time': 40,
+        'max_cells': 29,
+        'crop_size': 32,
+        'batch_size': 24,
+        'seed': 2,
+        'track_length': 7,
+        'val_split': 0.2
+    }, {
+        'testcase_name': 'test_data_03',
+        'time': 50,
+        'max_cells': 15,
+        'crop_size': 32,
+        'batch_size': 10,
+        'seed': 4,
+        'track_length': 6,
+        'val_split': 0.1
+    }
+])
+class TrackingTests(test.TestCase, parameterized.TestCase):
+
+    def test_temporal_slice(self, time, max_cells, crop_size,
+                            batch_size, seed, track_length, val_split):
+        X, y = self.create_test_data(time, max_cells, crop_size)
+        sliced_X, sliced_y = tracking.temporal_slice(X, y, track_length)
+
+        # Test temporal dimension correct index and value
+        for key in sliced_X:
+            self.assertEqual(sliced_X[key].shape[0], track_length)
+        for key in sliced_y:
+            self.assertEqual(sliced_y[key].shape[0], track_length - 1)
+
+    def test_random_rotate(self, time, max_cells, crop_size,
+                           batch_size, seed, track_length, val_split):
+        X, y = self.create_test_data(time, max_cells, crop_size)
+
+        # Get appearance and centroid tensors
+        X_apps = X['appearances']
+        X_cents = X['centroids']
+
+        # Assert rotating does not mutate the shape of tensors
+        rotated_X, y = tracking.random_rotate(X, y, 180)
+        self.assertEqual(rotated_X['appearances'].shape, X_apps.shape)
+        self.assertEqual(rotated_X['centroids'].shape, X_cents.shape)
+
+        # Assert rotating by 0 does not change tensors
+        X_apps = rotated_X['appearances']
+        X_cents = rotated_X['centroids']
+
+        output_X, y = tracking.random_rotate(rotated_X, y, 0)
+
+        self.assertAllEqual(output_X['appearances'], X_apps)
+        self.assertAllEqual(output_X['centroids'], X_cents)
+
+    def test_random_translate(self, time, max_cells, crop_size,
+                              batch_size, seed, track_length, val_split):
+        X, y = self.create_test_data(time, max_cells, crop_size)
+
+        # Get initial centroid values for two different cells
+        init_x_0 = X['centroids'][0, 0, 0]
+        init_x_1 = X['centroids'][0, 1, 0]
+
+        translated_X, y = tracking.random_translate(X, y)
+
+        # Get distance translated for two different cells
+        r0 = (float)(translated_X['centroids'][0, 0, 0] - init_x_0)
+        r1 = (float)(translated_X['centroids'][0, 1, 0] - init_x_1)
+        # Assert that centroids are translated by the same amount
+        assert abs(r1 - r0) < 0.001
```

Comment:
Not a bad start, but a few things I think we can improve:

* The tolerance of 1e-3 seems rather arbitrary. Shouldn't they get translated the exact same amount?
* We can make this assertion for all points in the array, not just 2 random ones.

Here's a quick mockup of how I think these can be addressed:

```python
translated_X, y = tracking.random_translate(X, y)

diff = X['centroids'] - translated_X['centroids']

# each value should be translated the same amount
single_value = np.empty_like(diff)
single_value.fill(diff[0, 0, 0])
self.assertAllEqual(diff, single_value)
```